### PR TITLE
GHA: add ArborX build

### PR DESCRIPTION
**Pull request by junghans**
Part of https://github.com/ECP-copa/Cabana/issues/345

### CODE REVIEW

**Comment by streeve** [comment]
Looks like the ArborX build failed

File: .github/workflows/CI2.yml [hunk 1]
```diff
@@ -108,14 +115,32 @@ jobs:
...
+      - name: Checkout arborx
+        if: ${{ matrix.arborx }}
+        uses: actions/checkout@v2.2.0
+        with:
+          repository: arborx/ArborX
+          ref: master
+          path: arborx
+      - name: Build arborx
+        if: ${{ matrix.arborx }}
+        working-directory: arborx
```
Comment (streeve) [inline .github/workflows/CI2.yml:127]: Want to fix my `cd build`?

File: .github/workflows/CI2.yml [hunk 2]
```diff
@@ -108,14 +115,32 @@ jobs:
...
+        run: |
+          cmake -B build \
+            -DCMAKE_PREFIX_PATH=${HOME}/kokkos
```
Comment (dalg24) [inline .github/workflows/CI2.yml:130]:
```suggestion
            -DKokkos_ROOT=${HOME}/kokkos
```

File: .github/workflows/CI2.yml (hunk 1 above)
Comment (junghans) [reply .github/workflows/CI2.yml:127]: `cmake -B build` does that, is was a missing `\`

File: .github/workflows/CI2.yml (hunk 2 above)
Comment (junghans) [reply .github/workflows/CI2.yml]: Does it matter? Both should do the same.

Comment (dalg24) [reply .github/workflows/CI2.yml]: It does not matter.  Just my preference ;)

**Comment by streeve** [comment]
We'll have to pick a different distribution for now. And do we want to require the ArborX build to pass?

File: .github/workflows/CI2.yml [hunk 3]
```diff
@@ -20,69 +20,88 @@ jobs:
...
             openmp: 'ON'
             cmake_build_type: 'Release'
             kokkos_ver: 'master'
+            arborx: ['NoArborX']
+          - distro: 'ubuntu:latest'
+            cxx: 'g++'
+            openmp: 'ON'
+            cmake_build_type: 'Release'
+            kokkos_ver: 'master'
+            arborx: 'ArboX'
```
Comment (streeve) [inline .github/workflows/CI2.yml:104]: 'ArborX'

**Review by streeve — APPROVED** [review]

**Review by streeve — APPROVED** [review]

**Comment by junghans** [comment]
@streeve do we know why the intel builds are failing?

**Comment by streeve** [comment]
> @streeve do we know why the intel builds are failing?

Same as the other distributions (periodic `Halo`)

**Pull request merged**